Show modal warning users when document fetch results in some errors

amprokop opened the pull request:
Shows modal for the error case when some documents couldn't be fetched.

Fixes #160.

![final_efolder_modal](https://cloud.githubusercontent.com/assets/3781835/19520903/c5f5c58a-95df-11e6-97f0-a05c987ee5ac.gif)

@lakohl — added this GIF to give you an idea of what it currently looks like, and to obtain real content. wanna get that to me when you can?

Test plan: fake a download error 

```
<div class="cf-app-segment cf-app-segment--alt">
  <% if @download.complete? %>
    <% if true %>
```

tab through, check that focus is trapped
check that modal info is marked by aria-labelledby and aria-describedby
check that esc key closes modal
check that focus returns to the previously focused element when modal is closed


cc @KHarshawat

amprokop commented:
@shanear — waiting on content from @lakohl @abbyraskin on this, but should be ready for review. thanks!

amprokop commented:
@shanear — just noticed that CI fails a check. weird bc it passes on my machine. i'll take a look, but the application code itself should still be ready for review.

shanear commented:
@amprokop my gut tells me it could have to do with a version discrepancy with PhantomJS

amprokop commented on spec/features/download_spec.rb:
```diff
@@ -293,6 +293,36 @@
+    end
+
+    # check that focus is correctly moved to the modal
+    is_focus_on_modal = page.evaluate_script("document.activeElement.className").include? "cf-modal-startfocus"
```
@joofsh — this is the test i was considering removing

amprokop commented on spec/features/download_spec.rb:
```diff
@@ -293,6 +293,36 @@
+
+    # check that focus is correctly moved back to the "Download Anyway" button
+    # after the modal is closed
+    is_focus_on_button = page.evaluate_script("document.activeElement.className").include? "cf-action-openmodal"
```
@joofsh @shanear — i also propose to remove this test

amprokop commented:
interesting. I updated PhantomJS and removed the tests relating to focused elements — and the Travis CI build still fails. It looks like the modal isn't popping up in the phantomJS browser maybe. More to investigate.

lakohl commented:
@amprokop here copy for the modal:
We encountered errors when retrieving some documents and they won’t be included in the eFolder download. If you elect to 'Download anyway', you may want to retrieve these files individually from VBMS.

amprokop commented:
@lakohl — one more question too. The content above ^^ uses single quotes. I assume we want double quotes for this, right?

amprokop commented on app/assets/javascripts/application/modal.js:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    if (tabKey) {
+      if ($('.cf-modal-endfocus').is(':focus')) {
```
ahh this actually isn't scoped, you're right

amprokop commented:
@shanear — i made some changes to scope this more tightly. going to update from master, retest, then merge since you approved this.

amprokop commented:
![image](https://cloud.githubusercontent.com/assets/3781835/19698099/ef6df172-9abb-11e6-94ed-e90abf711f52.png)
